### ISSUE TITLE
One hot encoding, multicollinearity, dummy trap

### ISSUE THREAD

**Issue by RobertSamoilescu** [SeldonIO/alibi#718]
A standard practice when transforming the label encoding to one-hot encoding (ohe) is to drop the first column to address multicollinearity (dummy trap). Currently there exist some explainers that know how to deal with one-hot encoded features such as: [AnchorTabular](https://github.com/SeldonIO/alibi/blob/e790f52a2b19236cb3beb8be90962f2116057f8d/alibi/explainers/anchors/anchor_tabular.py#L589), [Counterfactuals](https://github.com/SeldonIO/alibi/blob/e790f52a2b19236cb3beb8be90962f2116057f8d/alibi/explainers/cfproto.py#L47). Some utility function used by those are available here. The limitation of using the `ohe` flag (i.e. passing ohe dataset) is that the explainers don't know how to deal with the case where the ohe uses `drop='first'` (see [sklearn docs](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html)), which seems to be a standard procedure/best practice. 

It would be good to see if we can extend the capabilities of our explainers and utility functions to deal with all the arguments available in [sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html). Otherwise, it would be preferable to mention explicitly in our documentation what "format" the ohe is expected to be in (i.e. full representation).
